src/io.jl

```diff
@@ -82,3 +137,143 @@ function write_clustering_result_to_tables(
 
   return nothing
 end
+
+"""
+A helper function to compute the rep_period offset for group indexing.
+"""
+function _rep_period_offset(num_rep_periods::Int, group_idx::Int)
+  return num_rep_periods * (group_idx - 1)
+end
+
+"""
+A function to offset representative period indices so that groups have disjoint rep_period ranges.
+For group index g, new_rep_period = old_rep_period + offset given by _rep_period_offset(n_rp, g).
+"""
+function _combine_group_profiles(
+  results::Dict{
+    DataFrames.GroupKey{GroupedDataFrame{DataFrame}},
+    TulipaClustering.ClusteringResult,
+  },
+  n_rp::Int,
+)
+  profile_dfs = DataFrame[]
+  for (g, (_, group_result)) in enumerate(pairs(results))
+    df = group_result.profiles === nothing ? DataFrame() : copy(group_result.profiles)
+    if !isempty(df)
+      df.rep_period .+= _rep_period_offset(n_rp, g)
+    end
+    push!(profile_dfs, df)
+  end
+  return isempty(profile_dfs) ? DataFrame() : vcat(profile_dfs...; cols = :union)
+end
+
+"""
+A function to combine weight matrices from different groups.
+For group index g, new_rep_period = old_rep_period + offset given by _rep_period_offset(n_rp, g).
+"""
+function _combine_weight_matrices(
+  results::Dict{
+    DataFrames.GroupKey{GroupedDataFrame{DataFrame}},
+    TulipaClustering.ClusteringResult,
+  },
+  n_rp::Int;
+  layout = ProfilesTableLayout(),
+)
+  weight_matrices_dfs = Vector{DataFrame}(undef, length(results))
+  year_col = layout.year
+
+  for (g, (group_key, group_result)) in enumerate(pairs(results))
+    df = weight_matrix_to_df(group_result.weight_matrix)
+    df.rep_period .+= _rep_period_offset(n_rp, g)
+
+    if year_col in keys(group_key)
+      year_value = group_key[year_col]
+      insertcols!(df, 1, year_col => year_value)
+    end
```

Comment:
From pair review. We can include all columns that are in group_key instead. The extra columns are ignored by Tulipa.